src/core/math/color.js

```diff
@@ -254,6 +275,30 @@ class Color {
         return s;
     }
 
+    /**
+     * Converts the color to an array of numbers.
+     *
+     * @param {boolean} [alpha] - If true, the output array will include the alpha value.
+     * @param {number[]} [out] - The array to populate with the color components. If not specified,
+     * a new array is created.
+     * @param {number} [offset] - The zero-based index at which to start copying elements to the
+     * array. Default is 0.
+     * @returns {number[]} The color as an array of numbers.
+     * @example
+     * const c = new pc.Color(1, 1, 1);
+     * // Outputs [1, 1, 1, 1]
+     * console.log(c.toArray(true));
+     */
+    toArray(alpha, out = [], offset = 0) {
```

Comment:
I'm not keen on `alpha` being the first parameter here. It would be nice to have symmetry for arguments for `fromArrray` and `toArray` (e.g. first arg is `array` and second arg is `offset`). Then you could add `alpha` as the third arg for `Color`.

Reply:
Also (super minor thing buuuut...), share the same name for the array arg for both `fromArray` and `toArray`: `array` or `arr` maybe. But not keen on `out`.